### ISSUE TITLE
Battleground: WSG - Another LoS issue

### ISSUE THREAD

**Issue by iluminatix** [SunwellTracker/issues#2185]
Decription:  you cant hide behind threes or the stumps in midfield (and other places) inside wsg, hunters, or other ranged attackers just attack through the "pillar" 

How it works: 
you hide behind LoS, there is still LoS on you from any ranged attacker, and you cant hide.

How it should work: 
You hide behind LoS, ranged attacker can not attack you

Source (you should point out proofs of your report, please give us some source): 
go in bg and test it, i really cba to make a video of this, but its so easy for you to test. 

- Best regards
 hope you fix, its really gamebreaking when random BM scrub can just pve you down cus there is no LoS in wsg as there should be

**Comment by Akha72** [comment]
Well, in fact you can. Some places don't work and one can cast through them, while others work too well and make you 'not in the line of sight' except if someone stands in the exact same pixel spot. Many of those things were reported back when the old bugtracker was open, with not much luck tbh. The more pressing issue however is the fact that you can melee through textures (walls, trees, fences, pillars) if your target is within melee yard range. This is really gamebreaking and was reported too.

**Comment by iluminatix** [comment]
@Akha72 first of all... the fact that there are other issues, does not mean this one is less important imo, the importance rank all depends what class you play i guess, since it infuence how either bug anoys you. I mainly play druid (and other ranged classes) and i can say for sure that its much more gamebreaking to me that i cant hide behind what seems like an obvious (and btw well known, hiding from LoS place)
than it is that a melee can auto attack me through the wall if in melee range. its so much worse that a hunter can shoot me through hiding places using every availeble ability!! its so shit tbh... 

with that said, i know there are other issue with LoS in wsg, i did a search before i made this post, to check if it was already posted, and saw other issue you mention, thats why i called the title "Another LoS issue"

on the issue that you mention about attacking through texture though, i feel like its been like this everywhere i played (cant remember for retail@2007), and it never bothered me much, since i can still hide... and they can never kill me with auto attack only.. what server had this logic? i havent seen it and i seen many :)

**Comment by iluminatix** [comment]
will it be lazy / incompitent scripters on sw? like on feenix where chess event still nok working on 10th anniversary, i hope not, i still have faith

**Comment by Akha72** [comment]
Yeah, I wrote that the issue of melee strikes going through walls is of more importance since you can do that in arena here and that's top tier of pvp experience (and los works more less fine in arenas). With los in bgs it's a little bit complex since not every tree is supposed to be an obstruction and to prove it is or it isn't one would need some retail backup from 3.3.5a since various los elements has changed in bgs over time.

**Comment by Maadguy** [comment]
Your confusion is understandable, but this is in fact Blizzlike, tho most people hate it.

Basically, you need to understand the difference between M2, and WMO. WMO would be something like a building, M2 would be something like a tree.

M2's are not supposed to create a LOS, which is super confusing, because some trees for exemple, are M2's, and some are WMO's, making it super hard to know which should create a LOS and which shouldn't. This is the case in warsong gulch :) 

Personnally, I wish they would allow M2's to create LOS, it's a simple fix, but yeah

**Comment by Aridam** [comment]
As said above, M2s (some trees and rocks etc., _not_ all, 'cause some are WMOs) are supposed to not obstruct LoS - - it's Blizzlike. This happens even with the land itself. 

Leaving this as-is would be Blizzlike, but many people are opposed to it (let's face it, not the first time, with the recent Random Dungeon Cooldown and how it got a lotta flak being an example).
